smrf/distribute/solar.py

```diff
@@ -268,38 +278,36 @@ def distribute(self, t, cloud_factor, illum_ang, cosz, azimuth,
                 :mod:`smrf.envphys.radiation.sunang`
             azimuth: azimuth to the sun for the basin, from
                 :mod:`smrf.envphys.radiation.sunang`
-            min_storm_day: decimal day of last storm for the entire basin, from
-                :mod:`smrf.distribute.precip.ppt.last_storm_day_basin`
             albedo_vis: numpy array for visible albedo, from
                 :mod:`smrf.distribute.albedo.Albedo.albedo_vis`
             albedo_ir: numpy array for infrared albedo, from
                 :mod:`smrf.distribute.albedo.Albedo.albedo_ir`
 
         """
 
-        self._logger.debug('%s Distributing solar' % t)
+        self._logger.debug(f'{date_time} Distributing solar')
 
         # Only calculate solar if the sun is up
         if cosz > 0:
             self.cloud_factor = cloud_factor.copy()
 
-            wy_day, wyear, tz_min_west = self.radiation_dates(t)
-
             # --------------------------------------------
             # calculate clear sky radiation
 
             # Not all the clean but it will work for now
-            val_beam, val_diffuse = self.calc_ir(min_storm_day, wy_day,
-                                                 tz_min_west, wyear, cosz,
-                                                 azimuth)
+            val_beam, val_diffuse = self.calc_stoporad(
+                date_time, illum_ang, cosz, azimuth,
+                albedo_ir, self.IR_WAVELENGTHS)
+
             setattr(self, 'clear_ir_beam', val_beam)
             setattr(self, 'clear_ir_diffuse', val_diffuse)
             self.ir_beam = val_beam.copy()
             self.ir_diffuse = val_diffuse.copy()
 
-            val_beam, val_diffuse = self.calc_vis(min_storm_day, wy_day,
-                                                  tz_min_west, wyear, cosz,
-                                                  azimuth)
+            val_beam, val_diffuse = self.calc_stoporad(
+                date_time, illum_ang, cosz, azimuth,
+                albedo_vis, self.VISIBLE_WAVELENGTHS)
+
             setattr(self, 'clear_vis_beam', val_beam)
             setattr(self, 'clear_vis_diffuse', val_diffuse)
```

Comment:
This is mainly observational, and I like to use `@property` decorator in classes and then have a more readable code:
https://pythonguide.readthedocs.io/en/latest/python/property.html#property

Reply:
So `calc_stoporad` would be a property? I'm not all that familiar with the `@property`

Reply:
yup, the `@decorator` is basically a wrapper for the above call.

Would look something like this:
```python
class Solar():
    def __init__(self, ...):
        self. calc_stoporad = value_from_init # Can also be `None` if not given with init

    @propert
    def calc_stoporad(self):
        return self. _calc_stoporad

    @ calc_stoporad.setter
    def calc_stoporad(self, val):
        self._radius = val
```